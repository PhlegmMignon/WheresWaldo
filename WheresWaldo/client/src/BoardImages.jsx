import convention from "./assets/nMpQXwq.jpg";
import convention_kon from "./assets/kon.webp";
import convention_luffy from "./assets/luffy.png";
import convention_zim from "./assets/Zim.yelling.svg";

import stage2 from "./assets/stage3.png";
import stage2_fish from "./assets/fish.png";
import stage2_cannon from "./assets/cannon.png";
import stage2_egg from "./assets/egg.png";

export default function BoardImages() {
  const getImages = () => {
    return [
      {
        name: "convention",
        src: convention,
        id: 0,
        characters: [
          { name: "zim", src: convention_zim, id: 0 },
          { name: "luffy", src: convention_luffy, id: 1 },
          { name: "kon", src: convention_kon, id: 2 },
        ],
      },
      {
        name: "stage 2",
        src: stage2,
        id: 1,
        characters: [
          { name: "fish", src: stage2_fish, id: 0 },
          { name: "cannon", src: stage2_cannon, id: 1 },
          { name: "egg", src: stage2_egg, id: 2 },
        ],
      },
    ];
  };

  return { getImages };
}
