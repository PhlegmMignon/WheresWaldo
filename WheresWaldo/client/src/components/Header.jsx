import { useEffect } from "react";

export default function Header({ gameState, characters, found, time }) {
  useEffect(() => {
    console.log(found);
  }, [gameState, found]);

  const renderHeader = (state) => {
    switch (state) {
      case "start":
        return (
          <div className="bg-yellow-400 text-2xl text-center text-gray-900 p-6">
            Pick a stage!
          </div>
        );

      case "inProgress":
        return (
          <div className="flex justify-center items-center gap-6 bg-yellow-400 text-2xl text-center text-gray-900 p-2">
            <div>Find the characters! </div>
            <li className="flex justify-between gap-6">
              {characters.map((char, index) => {
                return (
                  <ul key={char.id} className="text-lg">
                    <img
                      className="h-16"
                      src={char.src}
                      alt={"picture of" + char.name}
                    />
                    {found[index] ? (
                      <div className="line-through">{char.name}</div>
                    ) : (
                      <div>{char.name}</div>
                    )}
                  </ul>
                );
              })}
            </li>
            <div>
              {time.minutes}mins {time.seconds}s
            </div>
          </div>
        );

      case "end":
        return (
          <div className="flex justify-center items-center gap-6 bg-yellow-400 text-2xl text-center text-gray-900 p-2">
            <div>You found everyone! </div>
            <li className="flex justify-between gap-6">
              {characters.map((char, index) => {
                return (
                  <ul key={char.id} className="text-lg">
                    <img
                      className="h-16"
                      src={char.src}
                      alt={"picture of" + char.name}
                    />
                    {found[index] ? (
                      <div className="line-through">{char.name}</div>
                    ) : (
                      <div>{char.name}</div>
                    )}
                  </ul>
                );
              })}
            </li>
            <div>
              {time.minutes}mins {time.seconds}s
            </div>
          </div>
        );
    }
  };

  return <>{renderHeader(gameState)}</>;
}

//Make timer after DuringGame. Try to make timer only update once a second
//Check if found crossthrough works later
