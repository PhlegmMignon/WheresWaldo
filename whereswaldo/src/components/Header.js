import React from "react";

export default function Header(props) {
  return (
    <div id="header">
      <div id="container">
        <div id="char1Box"></div>
        <div id="char2Box"></div>
        <div id="char3Box"></div>
      </div>
      <div id="found?">
        <div id="found1"></div>
        <div id="found2"></div>
        <div id="found3"></div>
      </div>
      {props.timer}
    </div>
  );
}
