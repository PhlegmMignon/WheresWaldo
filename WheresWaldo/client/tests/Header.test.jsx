import { render } from "@testing-library/react";
import Header from "../src/components/Header.jsx";
import BoardImages from "../src/BoardImages.jsx";

describe("Header unit tests", () => {
  it("Renders stageSelect's header", async () => {
    const { container } = render(<Header gameState={"start"} />);

    expect(container).toMatchSnapshot();
  });

  it("Renders header during game", async () => {
    let gameImage = BoardImages().getImages();
    const chars = gameImage[0].characters;

    const { container } = render(
      <Header
        gameState={"inProgress"}
        characters={chars}
        found={[false, false, false]}
        time={{ minutes: 2, seconds: 4 }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("Renders header post game", async () => {
    let gameImage = BoardImages().getImages();
    const chars = gameImage[0].characters;

    const { container } = render(
      <Header
        gameState={"end"}
        characters={chars}
        found={[true, true, true]}
        time={{ minutes: 2, seconds: 4 }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
