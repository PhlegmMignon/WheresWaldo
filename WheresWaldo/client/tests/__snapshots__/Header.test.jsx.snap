// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header unit tests > Renders header during game 1`] = `
<div>
  <div
    class="flex justify-center items-center gap-6 bg-yellow-400 text-2xl text-center text-gray-900 p-2"
  >
    <div>
      Find the characters! 
    </div>
    <ul
      class="flex justify-between gap-6"
    >
      <li
        class="text-lg"
      >
        <img
          alt="picture ofzim"
          class="h-16"
          src="/src/assets/Zim.yelling.svg"
        />
        <div>
          zim
        </div>
      </li>
      <li
        class="text-lg"
      >
        <img
          alt="picture ofluffy"
          class="h-16"
          src="/src/assets/luffy.png"
        />
        <div>
          luffy
        </div>
      </li>
      <li
        class="text-lg"
      >
        <img
          alt="picture ofkon"
          class="h-16"
          src="/src/assets/kon.webp"
        />
        <div>
          kon
        </div>
      </li>
    </ul>
    <div>
      2
      mins 
      4
      s
    </div>
  </div>
</div>
`;

exports[`Header unit tests > Renders header post game 1`] = `
<div>
  <div
    class="flex justify-center items-center gap-6 bg-yellow-400 text-2xl text-center text-gray-900 p-2"
  >
    <div>
      You found everyone! 
    </div>
    <li
      class="flex justify-between gap-6"
    >
      <ul
        class="text-lg"
      >
        <img
          alt="picture ofzim"
          class="h-16"
          src="/src/assets/Zim.yelling.svg"
        />
        <div
          class="line-through"
        >
          zim
        </div>
      </ul>
      <ul
        class="text-lg"
      >
        <img
          alt="picture ofluffy"
          class="h-16"
          src="/src/assets/luffy.png"
        />
        <div
          class="line-through"
        >
          luffy
        </div>
      </ul>
      <ul
        class="text-lg"
      >
        <img
          alt="picture ofkon"
          class="h-16"
          src="/src/assets/kon.webp"
        />
        <div
          class="line-through"
        >
          kon
        </div>
      </ul>
    </li>
    <div>
      2
      mins 
      4
      s
    </div>
  </div>
</div>
`;

exports[`Header unit tests > Renders stageSelect's header 1`] = `
<div>
  <div
    class="bg-yellow-400 text-2xl text-center text-gray-900 p-6"
  >
    Pick a stage!
  </div>
</div>
`;
